src/Components/Drives/drives.ts

```diff
@@ -49,64 +49,58 @@ const drivesToElements = async (drives: Drive[]): Promise<string> => {
-	const data = await Storage.get('sidebar'); // Get user favorites data on sidebar
+const writeSidebarDriveItems = async (): Promise<void> => {
 	const drives = driveInfo.DRIVES;
-	if (!drives.length || platform === 'darwin') return `<div class="sidebar-nav-item" id="sidebar-drives"></div>`;
-	// Return basic sidebar item element if there's no drives detected or its running on macOS
-	else {
-		let drivesElement = '';
-		for (const drive of drives) {
-			//prettier-ignore
-			const driveName = platform === 'win32'
-			//prettier-ignore
-				? `${drive.name && /[^?]/.test(drive.name)
-					? drive.name
-					: drive.disk_type} (${drive.mount_point.replace(/\\$/g, '')})`
-			//prettier-ignore
-				: drive.mount_point.split('/')[drive.mount_point.split('/').length - 1]; // Get name of drive
-			drivesElement += `<span data-path = "${await escape(
-				drive.mount_point
-			)}" data-isdir="true" class="sidebar-hover-effect drive-item"><img src="${await fileThumbnail(
-				drive.is_removable ? 'usb' : 'hard-disk',
-				'favorites',
-				false
-			)}" alt="${driveName}"><span class="sidebar-text">${driveName}</span></span>`;
-		}
-		const result = `<div class="sidebar-nav-item sidebar-nav-drives ${data?.hideSection?.drives ? 'nav-hide-item' : ''}" id="sidebar-drives">
-			<div class="sidebar-hover-effect">
-			<span class="sidebar-nav-item-dropdown-btn" data-section="drives"><img src="${await fileThumbnail(
-				'hard-disk',
-				'favorites',
-				false
-			)}" alt="Drives icon"><span class="sidebar-text">${
-			platform === 'win32' ? await Translate('Drives') : await Translate('Pendrives')
-		}</span><div class="sidebar-nav-item-dropdown-spacer"></div></span>
-			</div>
-			<div class="sidebar-nav-item-dropdown-container">
-				${drivesElement}
-			</div>
-		</div>`;
-		return result;
+	const isWin32 = platform === 'win32';
+	const driveElement = document.querySelector<HTMLElement>('#sidebar-drives');
+	const driveBtnText = driveElement.querySelector('.sidebar-text');
+	driveBtnText.textContent = await Translate(isWin32 ? 'Drives' : 'Pendrives');
+	if (!drives.length || platform === 'darwin') {
+		driveElement.hidden = true;
+		return;
+	}
+	let content = '';
+	for (const drive of drives) {
+		let driveName: string;
+		if (platform === 'win32') {
+			const hasName = drive.name && /[^?]/.test(drive.name);
+			driveName = hasName ? drive.name : drive.disk_type;
+			driveName += ' (' + drive.mount_point.replace(/\\$/g, '') + ')';
+		} else driveName = drive.mount_point.split('/').at(-1);
+		const driveType = drive.is_removable ? 'usb' : 'hard-disk';
+		const iconPath = await fileThumbnail(driveType, 'favorites', false);
+		content +=
+			`<span data-path="${escape(drive.mount_point)}" data-isdir="true" class="sidebar-hover-effect sidebar-nav-item drive-item">\n` +
+			`  <div class="sidebar-icon">\n` +
+			`    <img src="${iconPath}">\n` +
+			`  </div>\n` +
+			`  <span class="sidebar-text">${driveName}</span>\n` +
+			`</span>`;
```

Comment:
@kimlimjustin My viewpoint is that if you make it clean enough, there's no need for 3rd-party frameworks like React. Wdy think?

Reply:
yes, true, I think we should refactor the code